source/postprocess/viscous_dissipation_statistics.cc

```diff
@@ -97,37 +103,33 @@ namespace aspect
       std::vector<double> viscous_dissipation (local_dissipation_integral.size());
       Utilities::MPI::sum (local_dissipation_integral, this->get_mpi_communicator(), viscous_dissipation);
 
-      const std::string unit = dim == 3 ? "(W)" : "(W/m)";
+      const std::string unit = (dim == 3) ? "(W)" : "(W/m)";
 
       // Add the dissipation per compositional fields to the statistics output
       for (unsigned int c = 0; c < n_compositional_fields; ++c)
         {
           statistics.add_value ("Viscous dissipation " + unit + " for composition " + this->introspection().name_for_compositional_index(c),
                                 viscous_dissipation[c]);
 
-          // also make sure that the other columns filled by the this object
+          // also make sure that the other columns filled by this object
           // all show up with sufficient accuracy and in scientific notation
-          const std::string columns[] = { "Viscous dissipation " + unit + " for composition " + this->introspection().name_for_compositional_index(c) };
-          for (unsigned int i=0; i<sizeof(columns)/sizeof(columns[0]); ++i)
```

Comment:
Yes, I don't remember who introduced it, but we're using this in a lot of places for no good reason :-(